option.go

```diff
@@ -149,6 +149,18 @@ func NewOption(kind reflect.Kind, names ...string) Option {
 }
 
 func (o *option) WithDefault(v interface{}) Option {
+	if v == nil {
+		panic(fmt.Errorf("cannot use nil as a default"))
+	}
+
+	// if type of value does not match the option type
+	if vKind, oKind := reflect.TypeOf(v).Kind(), o.Type(); vKind != oKind {
```

Comment:
This feels like it alleviates a large footgun stemming from Go's lack of generics. Especially since in go-ipfs we build the commands at compile time it seems like panicking when someone has improperly set the default is just a good idea and will save us from mistakes we've made in the past such as using an `1` instead of a `uint(1)` default value in `UintOption` or using `"foo"` instead of `[]string{"foo"}` as a default value in StringsOption.